product/v2.py

```diff
@@ -235,26 +236,36 @@ def ensure_is_v2(
         raise RuntimeError(f'unsupported component-descriptor-version: {schema_version=}')
 
 
+def _target_oci_repository_from_component_name(component_name: str, ctx_repo_base_url: str):
+    component_name = component_name.lower()
```

Comment:
nit: factor-out to a function `normalise-component-name` (including other occurrences)